### PR TITLE
Added a flow map parameter to HDRI Sky

### PR DESCRIPTION
**Pull request by adrien-de-tocqueville**
### Purpose of this PR

This PR adds a flow map option to distort the HDRI sky cubemap.
It can be a user provided flowmap, or a uniform flowmap generated from a wind direction.

sky

kirby2

Link to old PR: https://github.com/Unity-Technologies/ScriptableRenderPipeline/pull/6446

---
### Testing status

**Manual Tests**: What did you do?
- [x] Opened test project + Run graphic tests locally
- [x] Built a player
- [x] Checked new UI names with UX convention
- [x] Tested UI multi-edition + Undo/Redo + Prefab overrides + Alignment in Preset
- [x] C# and shader warnings (supress shader cache to see them)

**Automated Tests**: Added one graphics test (5009)

**Yamato**: (Select your branch):
https://yamato.prd.cds.internal.unity3d.com/jobs/902-Graphics/tree/HDRP%252Fhdri-sky-flow

### CODE REVIEW

**Comment by iM0ve** [comment]
Test status: in progress
Test doc: https://favro.com/organization/c564ede4ed3337f7b17986b6/8ddd69383a55181bd8da3bda?card=Uni-132337

File: com.unity.render-pipelines.high-definition/Runtime/Sky/SkyUtils.hlsl [hunk 1]
```diff
@@ -17,4 +17,14 @@ float3 GetSkyViewDirWS(float2 positionCS)
     return normalize(viewDirWS.xyz);
 }
 
+// Returns latlong coords from view direction
+float2 GetLatLongCoords(float3 dir, float upperHemisphereOnly)
+{
+    const float2 invAtan = float2(0.1591, 0.3183);
+    float2 uv = float2(atan2(dir.x, dir.z), asin(dir.y));
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/Sky/SkyUtils.hlsl:24]: Please use fastAtan / fastAsin (see common.Hlsl), also see if they have the required precision (try with a 4K sky to check that we don't get artifact)

Comment (adrien-de-tocqueville) [reply com.unity.render-pipelines.high-definition/Runtime/Sky/SkyUtils.hlsl]: I am not too sure about my use of fastAtan for atan2 but it works with minor precision difference

**Comment by iM0ve** [comment]
Test status: **In progress**
1) We should change the Flowmap and Procedural to be selectable from a drop down menu. As is, its not clear for the user that there is another option until procedural is unchecked

gcjytdurstr

**Pull request merged**